src/mimefactory.rs

```diff
@@ -826,11 +826,19 @@ impl<'a, 'b> MimeFactory<'a, 'b> {
 
         // add attachment part
         if chat::msgtype_has_file(self.msg.viewtype) {
-            if !is_file_size_okay(context, &self.msg) {
-                bail!(
-                    "Message exceeds the recommended {} MB.",
-                    RECOMMENDED_FILE_SIZE / 1_000_000,
-                );
+            let max_file_size = context.get_config_int(Config::MaxAttachSize) as u64;
+            if !is_file_size_okay(context, &self.msg, max_file_size) {
+                if max_file_size < RECOMMENDED_FILE_SIZE {
```

Comment:
i think, this should just not happen, if the max_file_size is 10mb, the recommended size should be adapted to that.

Reply:
Do you mean that recommended size should be a function of configured file size (config_value/2 - 375k)?

Reply:
It is also unclear for me why comparison happens with one size, but another size is reported to the user. I am not sure what was the motivation for introducing recommended size concept, but it seems like misleading and excessive for me.

Reply:
the "recommended size concept" is needed as it does not mean much if on a given account the user can send 100mb or more if many recipients are not able to receive this message.

Reply:
My understanding after your answer is, that recommended size needs to be a constant, if maximum file size is less than recommended size, then recommended size is not to be reported, otherwise message is to report max file size and recommended size. Anyway it looks strange to me that comparison is with the max file size (doesn't matter hard-coded or configured) but message is saying that file exceeds recommended size.